src/components/Form/UploadDocumentForm.jsx

```diff
@@ -171,7 +166,7 @@ const UploadDocumentForm = () => {
           </FormControl>
           <br />
           <FormControl fullWidth>
-            <Button variant="contained" disabled={state.processing} type="submit" color="primary">
+            <Button variant="contained" disabled={uploading || !file} type="submit" color="primary">
```

Comment:
I believe the `!file` condition is enough to trigger disabled?

Reply:
only when there's no file. While it's uploading that file it stays active until it's complete.  so the uploading state is necessary to trigger disabled while that is happening, it's replacing what used to be state.processing.

Reply:
I'm a but confused here.

Disabled is being triggered only when there's a file in the new state. Which makes sense. Condition doesn't clear until the file is cleared, which is good too since it's only triggered at the end of handler function or when clear forms are clicked.

So doesn't that make uploading redundant?

Reply:
No, a file in the new state does not disable the upload button. I'm not actually sure of any value that corresponds to a `new file` state. Upload is disabled in two cases:

1. The user has not attached a file to the form (`file` is null or undefined)
2. The form is currently performing an upload (`uploading` is true)

As long as both those cases are false (i.e. there is a file attached to the form, and the form is not currently uploading), the upload button is enabled.

Reply:
^ yep... The button should not stay enabled while the file is being uploaded, just allows the user to click it more times.  It should be disabled while uploading.    The issue is that the `file` state is still true while it is uploading, so it requires both conditions (`!file` and `uploading`) to solve.  That's what I was trying to describe in our DM as well.

Reply:
That should be fine then, I thought we're replacing state.processing completely.

I'm good if that's what was intended to keep functionality from state.processing.

I'm just curious now as to why we couldn't just use state.processing then. Like we could move the dispatch function to where the set function calls are for loading.

Reply:
have you tested?  It should, but I was having trouble with it. Like the state.processing stopped working or something. I was going to ask about it too.  I could never get state.processing to be true anymore.

Reply:
I'll give this method a shot and update.

Reply:
Updated this to use notification dispatch instead. As far as I can tell, though, the notification is global state, yes? So we're only able to set one `SET_PROCESSING` value for the whole system? That seems like an odd limitation.